tests/spec/core/webidl-spec.js

```diff
@@ -1077,6 +1091,41 @@ callback CallBack = Z? (X x, optional Y y, /*trivia*/ optional Z z);
     const enumValueType = doc.getElementById("dom-enumtype-enumvaluetype");
     expect(enumValueType.dataset.idl).toBe("enum-value");
   });
+  it("auto-links based on definition context", async () => {
+    const body = `
+      <section>
+        <h2>Test</h2>
+        <pre class="idl" id="link-test" data-cite="HTML DOM">
+          interface Foo {
+            DOMString fromWebIDL(); // defined in WebIDL
+            attribute EventTarget fromDomSpec; // Defined in DOM
+            attribute EventHandler fromHTMLSpec; // Defined in HTML
+          };
+        </pre>
+      </section>
+    `;
+    const ops = makeStandardOps({ xref: true }, body);
+    const doc = await makeRSDoc(ops);
+
+    // DOMString fromWebIDL();
+    const domString = doc.querySelector(
+      "#link-test a[href='https://heycam.github.io/webidl/#idl-DOMString']"
+    );
+    expect(domString).toBeTruthy();
```

Comment:
Thinking of adding a custom jasmine matcher for this stuff `expect(selector).toExistInDocument(doc)` (or similar)

Reply:
something like that might be nice. However, it's pretty unusual that we just check if a node is truthy()... I was being lazy :)